doc/developer/design/20230512_mir_cost_model.md

```diff
@@ -151,3 +171,13 @@ How should we communicate source features to the cost model?
 [future-work]: #future-work
 
 When we have a candidate cost model, we will want to identify opportunities to use the cost model in the optimizer and pick a good first place to use it—most likely join planning. Actually using the cost model in the optimizer will require more validation and testing—at a minimum on LDBC, but ideally on client queries—to ensure that we don’t regress performance.
+
+We've identified the following early candidate clients for the cost model:
+
+1. **Join ordering.** There are some trade offs here between set enumeration and more heuristic approaches---in an ideal world, we'd have an explicit "optimization budget" to determine how much of the join ordering space to explore.
```

Comment:
As noted above, I don't think we need the throw-away `MIR ⇒ LIR` lowering for this use case.